packages/babel-parser/src/parser/expression.ts

```diff
@@ -3210,6 +3246,39 @@ export default abstract class ExpressionParser extends LValParser {
     return this.finishNode<N.ModuleExpression>(node, "ModuleExpression");
   }
 
+  parseVoidPattern(
+    this: Parser,
+    refExpressionErrors: ExpressionErrors | null,
+  ): N.VoidPattern {
+    this.expectPlugin("discardBinding");
+    const node = this.startNode<N.VoidPattern>();
+    if (refExpressionErrors != null) {
+      refExpressionErrors.voidPatternLoc = this.state.startLoc;
+    }
+    this.next();
+    return this.finishNode(node, "VoidPattern");
+  }
+
+  parseMaybeAssignAllowInOrVoidPattern(
+    this: Parser,
+    close: TokenType,
+    refExpressionErrors: ExpressionErrors | null,
+    afterLeftParse?: Function,
+  ) {
+    if (refExpressionErrors != null && this.match(tt._void)) {
+      const nextType = this.lookahead().type;
+      if (nextType === tt.comma || nextType === close || nextType === tt.eq) {
```

Comment:
```suggestion
      const nextCode = this.lookaheadCharCode();
      if (
        nextCode === charCodes.comma ||
        nextCode ===
          (close === tt.bracketR
            ? charCodes.rightSquareBracket
            : close === tt.braceR
              ? charCodes.rightCurlyBrace
              : charCodes.rightParenthesis) ||
        nextCode === charCodes.equalsTo
      ) {
```
This passes all the tests. Does this work?